zerver/webhooks/groove/view.py

```diff
@@ -0,0 +1,83 @@
+
+    # We listen to several events that are used for notifications.
+    # Other events are ignored.
+    if event == 'ticket_started':
+        body = u'New ticket from {customer_name}'
+        body += u"\n```quote\n**[Ticket #{number}: {title}]({app_url})**\n{summary}\n```"
+    elif event == 'ticket_assigned':
+        # Take the state, assignee, and assigned group from the payload.
+        state = payload['state']
+        assignee = payload['assignee']
+        assigned_group = payload['assigned_group']
+
+        # There are three states on a ticket: opened,
+        # pending, and closed. This creates the message
+        # based on the state of the ticket.
+        if state == "opened":
+            body = u"An open ticket has been assigned to"
+        else:
+            body = u"A {state} ticket has been assigned to"
+
+        # If there is a person and/or a group assigned,
+        # make a notification message. Otherwise, ignore it.
+        if assignee or assigned_group:
+            if assignee and assigned_group:
+                body += u" {assignee} from {assigned_group}"
+            elif assignee:
+                body += u" {assignee}"
+            elif assigned_group:
+                body += u" {assigned_group}"
+            body += u"\n```quote\n**[Ticket #{number}: {title}]({app_url})**\n```"
+        else:
+            return json_success()
+    elif event == 'agent_replied':
+        # Take the agent's email and the ticket number from the payload.
+        agent = payload['links']['author']['href'].split("http://api.groovehq.com/v1/agents/")[1]
+        number = payload['links']['ticket']['href'].split("http://api.groovehq.com/v1/tickets/")[1]
+
+        # Create the notification message.
+        body = u"%s has just replied to a ticket\n```quote\n**[Ticket #%s]" % (agent, number)
+        body += u"({app_ticket_url})**\n{plain_text_body}\n```"
+    elif event == 'customer_replied':
+        # Take the customer's email and the ticket number from the payload.
+        customer = payload['links']['author']['href'].split("http://api.groovehq.com/v1/customers/")[1]
+        number = payload['links']['ticket']['href'].split("http://api.groovehq.com/v1/tickets/")[1]
+
+        # Create the notification message.
+        body = u"%s has just replied to a ticket\n```quote\n**[Ticket #%s]" % (customer, number)
+        body += u"({app_ticket_url})**\n{plain_text_body}\n```"
+    elif event == 'note_added':
+        # Take the agent's email and the ticket number from the payload.
+        agent = payload['links']['author']['href'].split("http://api.groovehq.com/v1/agents/")[1]
+        number = payload['links']['ticket']['href'].split("http://api.groovehq.com/v1/tickets/")[1]
+
+        # Create the notification message.
+        body = u"%s has left a note\n```quote\n**[Ticket #%s]" % (agent, number)
+        body += u"({app_ticket_url})**\n{plain_text_body}\n```"
+    else:
+        return json_success()
+
```

Comment:
@infinitelooped: I would recommend moving all of the if/elif clauses to separate functions, one for each event. So you pass in the payload to each function, which returns the `body` as needed. See [how the GCI webhook does this](https://github.com/zulip/zulip/blob/master/zerver/webhooks/gci/view.py). The reason I am being a bit picky about this is that that kind of organization will be very helpful for adding more configurability to these webhooks in the not-too-distant future!